v3/spring-cloud-aws-v3-autoconfigure/src/main/java/io/awspring/cloud/v3/autoconfigure/properties/AwsSecretsManagerProperties.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.awspring.cloud.v3.autoconfigure.properties;
```

Comment:
Move this to: `io.awspring.cloud.v3.autoconfigure.secretsmanager` .
Please move `AwsSecretsManagerBootstrapConfiguration` to the package as well.
We need that because of client autoconfiguration.

Reply:
@MatejNedic Thanks for the review. 

I see that `AwsSecretsManagerBootstrapConfiguration` currently sets up the `AwsSecretsManagerPropertySourceLocator` and `AWSSecretsManager` beans. The `AwsSecretsManagerPropertySourceLocator` bean is no longer needed since this bean was  used only by legacy Spring Cloud Bootstrap phase based config loading. 

The `AWSSecretsManager` bean's equivalent in the AWS SDK V2 is the `SecretsManagerClient` which is already being registered in the `AwsSecretsManagerConfigDataLocationResolver` class. 

```
registerAndPromoteBean(resolverContext, SecretsManagerClient.class, this::createAwsSecretsManagerClient);
```
Thus the `SecretsManagerClient` bean will be available to all applicationContexts as the `bootstrapContext` is the parent(at which `SecretsManagerClient` got registered).

Due to the above reasons, I did not migrate the `AwsSecretsManagerBootstrapConfiguration` class. Please do let me know if you think there's something I might have overlooked.

Reply:
I would leave it because we can easily add `SecretsManagerTemplate` later with it. Tbh template is a great thing to have and we shouldn't force users to use `spring.config.import` to have a Bean for Client. What if someone just wants to use a template?  @eddumelendez I remember you were thinking about the SecretManagerTemplate so if you can share your thoughts.

Reply:
@MatejNedic IMHO I really like the concept of a `SecretsManagerTemplate`. And you have a valid point, on what if a user just wants a `SecretsManagerClient` without any desire of injecting config props into the environment. The use case being, some user trying to do operations like creating/deleteing secrets in AWS programmatically. 

The actual design of the `SecretsManagerTemplate` should however IMO be broader than just providing a wrapper over the `SecretsManagerClient`; we would most probably provide lifecycle methods underneath which alleviates the user of some boilerplate in their code. I have created a discussion #98 so that we unblock this one.

Reply:
@MatejNedic Please let me know if you still want me to retain the `SecretsManagerClient` bean creation auto configuration.